lib/amazon-pricing/helpers/instance-type.rb

```diff
@@ -154,10 +155,23 @@ def api_name_to_nf(name)
         size_to_nf[name.split('.').last]
       end
 
+      # note: the next smaller type may _not_ be supported for a given family
+      #  so this returns the next logical/possible smaller type, but not necessarily
+      #  the next valid type
       def next_smaller_type(name)
         fam,type = name.split('.')
-        nf= size_to_nf[type] / 2.0
-        new_type = NF_TO_SIZE_TABLE[nf] || NF_TO_SIZE_TABLE[nf.to_i] # 2.0 and 2 are no same when used as hash keys.
+        orig_nf = size_to_nf[type]
+        return nil unless orig_nf
+        # paranoia: assumes size_to_nf may not be sorted, which we need to step down
+        sorted_size_to_nf = {}
+        size_to_nf.sort_by(&:last).each_with_object({}) do |(size,nf)|
```

Comment:
This isn't a correct way to use `each_with_object`.
```
# either use each with object like
sorted_size_to_nf = size_to_nf.sort_by(&:last).each_with_object({}) do |(size,nf), h|
  h[size] = nf
end
# or just use each with the current way.
```

Reply:
yeah let's fix this

Reply:
changed to use 'each'